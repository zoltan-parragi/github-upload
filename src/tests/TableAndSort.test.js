import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BaseButton from '../components/Buttons';
import Table, {Sort} from '../components/TableAndSort';

Enzyme.configure({ adapter: new Adapter() });

describe('Table', () => {
	const props = {
		list: [
			{ title: '1', preview: 'x'},
			{ title: '2', preview: 'y'},
		],
		sortKey: 'TITLE',
		isSortReverse: false,
	};
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Table { ...props } />, div);
	});
	test('has a valid snapshot', () => {
		const component = renderer.create(
			<Table { ...props } />
	    );
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('shows two items in list', () => {
		const element = shallow(
			<Table { ...props } />
		);
		expect(element.find('.verse-display').length).toBe(2);
	});


});

describe('Sort', () => {
	const props = {
    	children: 'button',
	}
	it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<Sort {...props}/>, div);
	ReactDOM.unmountComponentAtNode(div);
	});
	test('has valid snapshot', () => {
		const component = renderer.create(<Sort {...props}/>);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
