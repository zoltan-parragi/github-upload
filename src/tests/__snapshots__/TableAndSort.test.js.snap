// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort has valid snapshot 1`] = `
<button
  className="button-sort button-sort-active"
  onClick={[Function]}
  type="button"
>
  button
</button>
`;

exports[`Table has a valid snapshot 1`] = `
<div
  className="table"
>
  <div
    className="sort"
  >
    <span
      className="sort-intro"
    >
      Sort on this page by 
    </span>
    <span>
      <button
        className="button-sort"
        onClick={[Function]}
        type="button"
      >
        reference
      </button>
    </span>
    <span
      className="sort-button-withmargin"
    >
      <button
        className="button-sort"
        onClick={[Function]}
        type="button"
      >
        verse text
      </button>
    </span>
    <span
      className="sort-button-withmargin"
    >
      <button
        className="button-sort button-sort-active"
        onClick={[Function]}
        type="button"
      >
        none
      </button>
    </span>
  </div>
  <div
    className="verse-display"
  >
    <span
      className="verse-display-reference"
    >
      1
    </span>
    <span
      className="verse-display-text"
    >
      x
    </span>
  </div>
  <div
    className="verse-display"
  >
    <span
      className="verse-display-reference"
    >
      2
    </span>
    <span
      className="verse-display-text"
    >
      y
    </span>
  </div>
</div>
`;
